dist/privacy-sandbox.js

```diff
@@ -0,0 +1,194 @@
+//[privacy-sandbox]
+/**
+ * Topics API
+ * Protected Audience API
+ * Attribution Reporting API
+ * Required command line flags: --enable-features=BrowsingTopics,InterestGroupStorage,PrivacySandboxAdsAPIsOverride
+*/
+
+let requests = $WPT_BODIES;
+const cannonicalFirstPartyDomain = getCanonicalDomain(document.location.hostname);
+
+let result = {
+  'topicsAPI': {
+    'topicsAvailable': document.featurePolicy.allowsFeature('browsing-topics'),
+    'usingBrowsingTopics': [],
+    'topicsAccessJs': [],
+    'topicsAccessHeader': [],
+    'observingTopics': []
+  },
+  'protectedAudienceAPI': {
+    'protectedAudienceAvailable': document.featurePolicy.allowsFeature('join-ad-interest-group'),
+    'interestGroups': {
+      'joinAdInterestGroup': [],
+      'leaveAdInterestGroup': [],
+      'updateAdInterestGroups': [],
+      'clearOriginJoinedAdInterestGroups': []
+    },
+    'runAdAuction': [],
+    'generateBid': [],
+    'scoreAd': [],
+    'reportWin': [],
+    'reportResult': []
+  },
+  'attributionReportingAPI': {
+    'attributionReportingAvailable': document.featurePolicy.allowsFeature('attribution-reporting'),
+    'attributionReportingEligibleHeader': {
+      'sentByBrowser': false,
+      'sentTo': [],
+    },
+    'completedRegistrations': {
+      'AttributionReportingRegisterSourceHeader': {},
+      'AttributionReportingRegisterTriggerHeader': []
+    },
+  },
+  attestationPublished: []
+}
+
+function getCanonicalDomain(hostname) {
+  const parts = hostname.split('.').reverse();
+  if (parts.length >= 2) {
+    return `${parts[1]}.${parts[0]}`;
+  } else {
+    return hostname;
+  }
+}
+
+async function fetchAndCheckAttestation(url) {
+  const controller = new AbortController();
+  const { signal } = controller;
+  setTimeout(() => controller.abort(), 5000);
+
+  try {
+    const response = await fetch(url, { signal, mode: 'no-cors' });
+    return response ? true : false;
+  } catch (error) {
+    return false;
+  }
+}
+
+let seenThirdParties = [];
+(async () => {
+  for (const request of requests) {
+    const url = new URL(request.full_url);
+    const isScript = request.type === 'Script';
+    const isDocument = request.type === 'Document';
+    const cannonicalRequestDomain = getCanonicalDomain(url.hostname);
+
+    let thirdPartyDomain = '';
+    // Maps all first party hostnames to the corresponding cannonical first party domain to truly consider only third-parties
+    if (cannonicalRequestDomain !== cannonicalFirstPartyDomain) {
+      thirdPartyDomain = url.hostname;
+    }
+    if (thirdPartyDomain && seenThirdParties.includes(thirdPartyDomain)) {
```

Comment:
So you're skipping the parsing for a domain after looking at a first request.
It's definitely helpful to avoid repeated attestations.
But as usually JS comes first it will skip following XHRs with headers.

Maybe move it more locally to attestation check?